apis/v1alpha2/shared_types.go

```diff
@@ -249,4 +238,90 @@ type RouteStatus struct {
 //
 // +kubebuilder:validation:MinLength=1
 // +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
 type Hostname string
+
+// Group refers to a Kubernetes Group. It must either be an empty string or a
+// RFC 1123 subdomain.
+//
+// This validation is based off of the corresponding Kubernetes validation:
+// https://git.k8s.io/apimachinery/pkg/util/validation/validation.go#L208
+//
+// Valid values include:
+//
+// * ""
```

Comment:
(empty string)

Reply:
Also mention that it implies core API group. Not a widely known thing AFAICT.

Reply:
Updated, thanks!